client/src/components/pages/personalProfile/index.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react'
+import { Form, Input } from 'antd'
+
+import Button from 'components/utils/Button'
+import './style.css'
+
+class PersonalForm extends React.Component {
+
+  state = {
+    confirmDirty: false,
+  };
+
+  componentDidMount = () => {
+    this.props.form.setFieldsValue(this.props.personal)
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.form.validateFieldsAndScroll((err, values) => {
+      if (!err) {
+        this.props.handlePersonalInfo(values, e)
+        console.log('Received values of form: ', values);
+      }
+    });
+  };
+
+  // location = this.props.location.state ? this.props.location.state.from : '/posts'
+
+  // handleCancel = (e) => {
+  //   e.preventDefault();
+  //   this.props.history.push(this.location)
+  // }
+
+  handleConfirmBlur = e => {
+    const value = e.target.value;
+    this.setState({ confirmDirty: this.state.confirmDirty || !!value });
+  };
+
+  compareToFirstPassword = (rule, value, callback) => {
+    const form = this.props.form;
+    if (value && value !== form.getFieldValue('password')) {
+      callback('Password must match!');
+    } else {
+      callback();
+    }
+  };
+
+  validateToNextPassword = (rule, value, callback) => {
+    const form = this.props.form;
+    if (value && this.state.confirmDirty) {
+      form.validateFields(['confirm'], { force: true });
+    }
+    callback();
+  };
+
+  render() {
+    const { getFieldDecorator } = this.props.form;
+
+    return (
+      <Form  onSubmit={this.handleSubmit} className='form'>
+
+        <Form.Item label="First Name" className='form--item'>
```

Comment:
@AbdallahAmmar96 It would be awesome in the future if this and the `register` form could share code, like the email / password fields and their validation logic.